memory/alloydb/chat_message_history.go

```diff
@@ -0,0 +1,224 @@
+package alloydb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+)
+
+type ChatMessageHistory struct {
+	engine     alloydbutil.PostgresEngine
+	sessionID  string
+	tableName  string
+	schemaName string
+	overwrite  bool
+}
+
+var _ schema.ChatMessageHistory = &ChatMessageHistory{}
+
+// NewChatMessageHistory creates a new NewChatMessageHistory with options.
+func NewChatMessageHistory(ctx context.Context, engine alloydbutil.PostgresEngine, tableName string, sessionID string, opts ...ChatMessageHistoryStoresOption) (ChatMessageHistory, error) {
+	var err error
+	// Ensure required fields are set
+	if engine.Pool == nil {
+		return ChatMessageHistory{}, errors.New("alloyDB engine must be provided")
+	}
+	if tableName == "" {
+		return ChatMessageHistory{}, errors.New("table name must be provided")
+	}
+	if sessionID == "" {
+		return ChatMessageHistory{}, errors.New("session ID must be provided")
+	}
+	cmh := ChatMessageHistory{
+		engine:    engine,
+		tableName: tableName,
+		sessionID: sessionID,
+	}
+	cmh, err = applyChatMessageHistoryOptions(cmh, opts...)
+	if err != nil {
+		return ChatMessageHistory{}, fmt.Errorf("applyChatMessageHistoryOptions(): %w", err)
```

Comment:
Need better error messages here (and the next one). Specifically something that gives the developer more context instead of just the method name -- e.g. something like "unable to apply provided options: %w" or "table schema is incorrect"